### PR TITLE
Convert OpenAPI to structured schema

### PR DESCRIPTION
**Pull request by lavalamp**

### CODE REVIEW

**Comment by lavalamp** [comment]
This now works on everything from the main schema, should be ready to go.

/approve

File: pkg/schemaconv/smd.go [hunk 1]
```diff
@@ -122,32 +162,34 @@ func (c *convert) VisitArray(a *proto.Array) {
...
+				c.reportError("missing map keys")
 			}
 		} else {
-			// TODO: emit warning?
+			c.reportError("unknown list type %v", val)
 			l.ElementRelationship = schema.Atomic
 		}
 	} else if val, ok := ext["x-kubernetes-patch-strategy"]; ok {
-		if val == "merge" {
+		if val == "merge" || val == "merge,retainKeys" {
```
Comment (apelisse) [inline pkg/schemaconv/smd.go]: You could maybe handle retainsKeys correctly and transform it into something useful

**Comment by apelisse** [comment]
> This now works on everything from the main schema, should be ready to go.

That's awesome

**Pull request merged**